### DISCUSSION TITLE
Calling SQL Functions (stored on Supabase) from a java client that uses Retrofit

### DISCUSSION THREAD

**Discussion by MaartensMathew** [supabase/supabase#29072] [Questions]
Hi there, I'm quite new to Supabase so please forgive me for incorrect terminology and understanding.

I have somewhat complex sql logic that I would like to offload to Supabase, in the form of a sql function that I can store and access later. I have built the SQL function (called getQuizEntries) that has 1 argument of a smallint and returns a table. The security is invoker level.

I am unsure how to access this function from the client side using Retrofit2. The baseURL is of the form :

"https://<database>.supabase.co"

I have been able to access the User table (for example) with the endpoint( I think that is what it's called ) of "/rest/v1/User". I was hoping that instead of referencing a table, I could reference my sql function that I have stored on my Supabase project. Google Gemini gives me something along the lines of "/rpc/<function-name" but I tried something like that but to no avail.

All my SupabaseAPI code logic is defined in a SupabaseAPI java interface, using @ annotations to shape the query. Might someone be able to guide me how to utilize these SQL functions (if possible)?

**Comment by GaryAustin1** [comment]
rpc call is the way.
If your function is in the public schema you can go to the dashboard API Docs tab and find your function and select BASH for the proper http call.
![image](https://github.com/user-attachments/assets/66c21e55-056c-40e6-a6b8-023513bb5bce)

**Reply by MaartensMathew** [reply]
Much appreciated, thank you. I hope you don't mind me asking 2 follow up questions.

1. I have a sql function that joins two tables and returns a table of entries. My argument (on supabase) is a single smallint (for int2), but on the client side, I had to make a whole class just for that one smallint (which i represented as a short) so that I would send through the class using the @body . Is there a way for me to send through arguments, without having to wrap using an @body ?

2. The project that I am using Retrofit2 in, is JavaFX Desktop and Android (Android Studio). In your opinion, do you think there are any functional differences between the ways that I would implement tools like the client, the supabase api interface, etc, in the javafx vs android? 

For example, I have the sql function: "https://<supabase-database>.co/rest/v1/rpc/getQuizEntries" and this is the one that works in JavaFX in the Supabase api interface. When i copied the interface over to android and attempted to run it, the error body of the response was:

"Failed to fetch data. Response code: 404 - Error body: {"error":"requested path is invalid"}

So based on this, I'm guessing Android and JavaFX handle things differently? Might you know of any things I should do to get the android version to work?
